lib/src/dartpad_support_server.dart

```
@@ -56,7 +52,63 @@ class FileRelayServer {
       throw (e);
     });
     _logger.info("Deleted Export with ID ${record.uuid}");
-    return new Future.value(new PadSaveObject.FromRecordSource(record));
+    return new Future.value(new PadSaveObject.fromRecordSource(record));
+  }
+
+  @ApiMethod(method: 'GET', path: 'getValidId')
+  Future<UuidContainer> getValidId() async {
+    int count = 0;
+    int limit = 4;
+    var database = ae.context.services.db;
+    String randomUuid;
+    var query;
+    List result;
+    do {
+      randomUuid = new uuid_tools.Uuid().v4();
+      query = database.query(_GistMapping)..filter('internalId =', randomUuid);
+      result = await query.run().toList();
+      count ++;
+    } while (!result.isEmpty && count < limit);
+    if (!result.isEmpty) {
+      _logger.severe("Could not generate valid ID.");
+      return new Future.value(new UuidContainer());
```

Please throw an exception here instead. We need to give an Err 500 back to the user